src/llm/backends/openai.py

```
@@ -17,6 +18,103 @@
+def _inject_schema_hint(
+    messages: list[dict[str, Any]],
+    response_format: type[BaseModel],
+) -> list[dict[str, Any]]:
+    """Append the target JSON schema to the system message.
+
+    ``json_object`` mode (unlike native ``json_schema``) does not tell the model
+    which fields to emit, so weaker / local models return empty or free-text
+    JSON. Injecting the schema makes the fallback reliable."""
+    try:
+        schema = json.dumps(response_format.model_json_schema())
+    except Exception as exc:
```

Ruff (0.15.17)

[warning] Do not catch blind exception: `Exception`

(BLE001)